pkg/controller/nstemplateset/nstemplateset_controller.go

```diff
@@ -207,30 +302,24 @@ func (r *NSTemplateSetReconciler) ensureUserNamespaces(logger logr.Logger, nsTmp
 }
 
 func (r *NSTemplateSetReconciler) ensureNamespace(logger logr.Logger, nsTmplSet *toolchainv1alpha1.NSTemplateSet, tcNamespace *toolchainv1alpha1.NSTemplateSetNamespace, userNamespace *corev1.Namespace) error {
-	username := nsTmplSet.GetName()
-
-	log.Info("provisioning namespace", "namespace", tcNamespace)
-	if err := r.setStatusProvisioning(nsTmplSet); err != nil {
+	log.Info("provisioning namespace", "namespace", tcNamespace.Type, "tier", nsTmplSet.Spec.TierName)
+	if err := r.setStatusProvisioning(nsTmplSet, fmt.Sprintf("provisioning the '-%s' namespace", tcNamespace.Type)); err != nil {
 		return err
 	}
-
-	params := map[string]string{"USERNAME": username}
-
 	if userNamespace == nil {
-		return r.ensureNamespaceResource(logger, nsTmplSet, tcNamespace, params)
+		return r.ensureNamespaceResource(logger, nsTmplSet, tcNamespace)
 	}
-	return r.ensureInnerNamespaceResources(logger, nsTmplSet, tcNamespace, params, userNamespace)
+	return r.ensureInnerNamespaceResources(logger, nsTmplSet, tcNamespace, userNamespace)
 }
 
-func (r *NSTemplateSetReconciler) ensureNamespaceResource(logger logr.Logger, nsTmplSet *toolchainv1alpha1.NSTemplateSet, tcNamespace *toolchainv1alpha1.NSTemplateSetNamespace, params map[string]string) error {
-	username := nsTmplSet.GetName()
-
+func (r *NSTemplateSetReconciler) ensureNamespaceResource(logger logr.Logger, nsTmplSet *toolchainv1alpha1.NSTemplateSet, tcNamespace *toolchainv1alpha1.NSTemplateSetNamespace) error {
+	logger.Info("creating namespace", "username", nsTmplSet.GetName(), "tier", nsTmplSet.Spec.TierName, "type", tcNamespace.Type)
```

Comment:
we could make the logging even better:
```suggestion
	logger = logger.WithValues( "username", nsTmplSet.GetName(), "tier", nsTmplSet.Spec.TierName, "type", tcNamespace.Type)
	logger.Info("creating namespace")
```
you coud do the same in the other methods